Automattic/jetpack
# Search: move search package constants to a class and remove `Jetpack` class usage

Fixes #23152; fixes #23149

#### Changes proposed in this Pull Request:
Move package version, installed path and slug to `Package` class, for `jetpack-autoloader` to load the latest version.

Note: there are still usage of `Jetpack` class, and they work as integrations with Jetpack and are left purposely.

#### Jetpack product discussion
pcNPJE-Ch-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Go to a site with Search subscription
* Ensure Instant Search, Search Dashboard, Customberg work well
* Ensure all tests pass
